pkg/asset/installconfig/gcp/usertags.go

```diff
@@ -0,0 +1,212 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	rscmgr "cloud.google.com/go/resourcemanager/apiv3"
+	rscmgrpb "cloud.google.com/go/resourcemanager/apiv3/resourcemanagerpb"
+	"github.com/googleapis/gax-go/v2/apierror"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+
+	"github.com/openshift/installer/pkg/types/gcp"
+)
+
+const (
+	// maxUserTagLimit is the maximum userTags that can be configured as defined in openshift/api.
+	// https://github.com/openshift/api/commit/ae73a19d05c35068af16c9aeff375d0b7c936a8a#diff-07b264a49084976b670fb699badaca1795027d6ea732a99226a5388104f6174fR604-R613
+	maxUserTagLimit = 50
+
+	// tagRequestEndPoint is the endpoint for tag requests.
+	tagRequestEndPoint = "https://cloudresourcemanager.googleapis.com"
+
+	// projectParentPathFmt is the format string for parent path of a project resource
+	projectParentPathFmt = "//cloudresourcemanager.googleapis.com/projects/%s"
+)
+
+// effectiveTags is the custom type for holding tags.
+type effectiveTags map[string]struct{}
+
+// tagManager handles resource tagging.
+type tagManager struct {
+	client API
+}
+
+// TagManager is the interface that wraps methods for resource tag operations.
+type TagManager interface {
+	getTagClientOptions() []option.ClientOption
+}
+
+// tagValuesClient handles operations on tag value resources.
+type tagValuesClient struct {
+	*rscmgr.TagValuesClient
+}
+
+// tagValuesClient handles operations on tag binding resources.
+type tagBindingsClient struct {
+	*rscmgr.TagBindingsClient
+}
+
+// NewTagManager creates a tagManager instance.
+func NewTagManager(client API) TagManager {
+	return &tagManager{client: client}
+}
+
+// ConvertUserTags converts tags from user provided list of tags in NamespacedName
+// `{parentID}/{tag_key_short_name}/{tag_value_short_name}` form to key:value pairs,
+// with key of the form `tagKeys/{tag_key_id}` and value of the form
+// `tagValues/{tag_value_id}`. Returns error when fetching a tag detail fails or when
+// tag already exists on the project resource.
+func ConvertUserTags(ctx context.Context, mgr TagManager, project string, userTags []gcp.UserTag) (map[string]string, error) {
+	if len(userTags) == 0 {
+		return nil, nil
+	}
+
+	if len(userTags) > maxUserTagLimit {
+		return nil, fmt.Errorf("more than %d tags is not allowed, configured: %d", maxUserTagLimit, len(userTags))
+	}
+
+	logrus.Debugf("user defined tags list: %v", userTags)
+
+	projectTags, err := getProjectTags(ctx, mgr, project)
+	if err != nil {
+		return nil, err
+	}
+
+	if dupTags := findDuplicateTags(userTags, projectTags); len(dupTags) != 0 {
+		return nil, fmt.Errorf("found duplicate tags, %v tags already exist on %s project", dupTags, project)
+	}
```

Comment:
This check should go in the installconfig asset validation

Reply:
comment incorporated.